docs/csharp/whats-new/csharp-9.md

```diff
@@ -0,0 +1,247 @@
+---
+title: What's new in C# 9.0 - C# Guide
+description: Get an overview of the new features available in C# 9.0.
+ms.date: 09/04/2020
+---
+# What's new in C# 9.0
+
+C# 9.0 adds the following features and enhancements to the C# language:
+
+- Records
+- Init only setters
+- Top-level statements
+- Pattern matching enhancements
+- Native sized integers
+- Function pointers
+- Suppress emitting localsinit flag
+- Target-typed new expressions
+- static anonymous functions
+- Target-typed conditional expressions
+- Covariant return types
+- Lambda discard parameters
+- Attributes on local functions
+- Module initializers
+- New features for partial methods
+
+C# 9.0 is supported on **.NET 5**. For more information, see [C# language versioning](../language-reference/configure-language-version.md).
+
+## Record types
+
+C# 9.0 introduces ***record types***, which are a reference type that provides synthesized methods to provide value semantics for equality. Records are immutable by default.
```

Comment:
I know that this has already been merged but I have concerns about the heavy emphasis on the immutability of "records" in C# when there really is little in the language that enforces or even encourages that they can't be mutated.  I feel that this emphasis will lead to a very false and confusing expectation.

Reply:
@HaloFour This is being discussed in #20601. Reading your comment, you may have better ideas for updating the description. I'll be watching that issue discussion before I make an update to this article.